code/game/machinery/bots/floorbot.dm

```diff
@@ -330,7 +334,7 @@ var/global/list/floorbot_targets=list()
 	// Needed because we used to look this up 15 goddamn times per process. - Nexypoo
 	var/list/shitICanSee = view(7, src)
 
-	if(src.amount <= 0 && !src.have_target())
+	if(src.amount < 50 && !src.have_target())

```

Comment:
Isn't that the point? It only gets that far down in process if he's not already repairing something. While idling (not repairing), he should search for tiles all the time, right?